File: 刷题/Hot 100/45. 跳跃游戏 II.go
func jump(nums []int) int {
    maxPos, end, step, n := 0, 0, 0, len(nums)
    for i:=0; i<n-1; i++ {
        if maxPos >= i {
            maxPos = max(maxPos, i+nums[i])
            if i == end {
                end=maxPos
                step++
            }
        }
    }
    return step
}

func max(a, b int) int {
    if a>b {
        return a
    }
    return b
}

File: 刷题/Hot 100/45. 跳跃游戏 II.cpp
class Solution {
public:
    int jump(vector<int>& nums) {
        int start=0;
        int end=1;
        int count=0;
        while(end<nums.size()){
            int maxnum=0;
            for(int j=start;j<end;j++){
                maxnum=max(maxnum, j+nums[j]);
            }   
            start=end;
            end=maxnum+1;
            count++;
        }
        return count;
    }
};